syntax = "proto3";
package nexus.hub.proto;

import "nexus/models/proto/typed_document.proto";
import "idm/api/proto/chat_manager_service.proto";

message StartDeliveryRequest {
  nexus.models.proto.TypedDocument typed_document = 1;
  idm.api.proto.Chat chat = 2;
  string bot_name = 3;
}

message StartDeliveryResponse {
  enum Status {
    OK = 0;
    TOO_MANY_DOWNLOADS = 1;
    ALREADY_DOWNLOADING = 2;
  }
  Status status = 1;
}

service Delivery {
    rpc start_delivery(StartDeliveryRequest) returns (StartDeliveryResponse) {};
}
